includes/User.php

```diff
@@ -2569,9 +2568,26 @@ public function getIntOption( $oname, $defaultOverride=0 ) {
 	 * @return string
 	 * @see getGlobalPreference
 	 */
-	public function getLocalPreference($pref, $cityId = null, $sep = "-", $default = null, $ignoreHidden = false) {
-		$globalPref = self::localToGlobalPropertyName($pref, $cityId, $sep);
-		return $this->getGlobalPreference($globalPref, $default, $ignoreHidden);
+	public function getLocalPreference($preference, $cityId = null, $sep = "-", $default = null, $ignoreHidden = false) {
+		global $wgPreferencesUseService;
+
+		if ($wgPreferencesUseService) {
+			$this->load();
+			$value = $this->userPreferences()->getLocalPreference($this->mId, $cityId, $preference, $default, $ignoreHidden);
+			wfRunHooks(
+				'UserGetPreference',
+				[
+					$this->userPreferences()->getPreferences($this->mId),
+					$preference,
+					&$value
+				]
+			);
```

Comment:
@drsnyder 
It came out that `UserGetPreference` hook need to be only used with global preferences, this is the only usage : https://github.com/Wikia/app/blob/48f8caf925b02009e9b20033eab437a7f972faf5/extensions/wikia/RTE/RTE.class.php#L680

But going back to your question, this hook is also called for `getOptionHelper` but inside the method : https://github.com/Wikia/app/blob/68ca84ba1e509d998893c6c883fbf3eb9b929bcc/includes/User.php#L2496